doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+
+### Commands
+
+#### `help`
+
+`help`
+
+Display the help message
+
+#### `version`
+
+`version`
+
+Display version info for the Windows Terminal
+
+#### `open-settings`
+
+`open-settings [--defaults,-d]`
+
+Open the settings file.
+
+**Parameters**:
+* `--defaults,-d`: Open the `defaults.json` file instead of the `profiles.json`
+  file.
+
+#### `list-profiles`
+
+`list-profiles [--all,-A] [--showGuids,-g]`
+
+Displays a list of each of the available profiles. Each profile displays it's
+name, seperated by newlines.
+
+**Parameters**:
+* `--all,-A`: Show all profiles, including profiles marked `"hidden": true`.
+* `--showGuids,-g`:  In addition to showing names, also list each profile's
+  guid. These GUIDs should probably be listed _first_ on each line, to make
+  parsing output easier.
+
+#### `new-tab`
+
+`new-tab [--initialPosition x,y]|[--maximized]|[--fullscreen] [--initialRows rows] [--initialCols cols] [terminal_parameters]`
+
+Opens a new tab with the given customizations. On it's _first_ invocation, also
+opens a new window. Subsequent `new-tab` commands will all open new tabs in the
+same window.
+
+**Parameters**:
+* `--initialPosition x,y`: Create the new Windows Terminal window at the given
+  location on the screen in pixels. This parameter is only used when initially
+  creating the window, and ignored for subsequent `new-tab` commands. When
+  combined with any of `--maximized` or `--fullscreen`, TODO: what do?
+* `--initialRows rows`: Create the terminal window with `rows` rows (in
+  characters). If omitted, uses the value from the user's settings. This
+  parameter is only used when initially creating the window, and ignored for
+  subsequent `new-tab` commands. When combined with any of `--maximized` or
+  `--fullscreen`, TODO: what do?
+* `--initialCols cols`: Create the terminal window with `cols` cols (in
+  characters). If omitted, uses the value from the user's settings. This
+  parameter is only used when initially creating the window, and ignored for
+  subsequent `new-tab` commands. When combined with any of `--maximized` or
+  `--fullscreen`, TODO: what do?
+* `[terminal_parameters]`: See [[terminal_parameters]](#terminal_parameters).
+
+
+#### `split-pane`
+
+`split-pane [--target,-t target-pane] [-h]|[-v] [--percent,-P split-percentage] [terminal_parameters]`
+
+<!-- `--percent,-P` is pretty close to `-p` from `--profile`. Is this okay? -->
+
+Creates a new pane by splitting the given pane vertically or horizontally.
+
+**Parameters**:
+* `--target,-t target-pane`: Creates a new split in the given `target-pane`.
+  Each pane has a unique index (per-tab) which can be used to identify them. If
+  omitted, defaults to `0` (the first pane).
+* `-h`, `-v`: Used to indicate which direction to split the pane. `-v` is
+  "vertically" (think `[|]`), and `-h` is "horizontally" (think `[-]`). If
+  omitted, defaults to vertical. If both `-h` and `-v` are provided, defaults to
+  vertical.

```

Comment:
As an overall philosophy, why default to vertical when both are provided as opposed to throwing an error?

Reply:
This kinda touches on something @miniksa & @leonMSFT have mentioned - when there's an error case like this, people seem to prefer "display an error" / "throw an exception". If there's an error case like this, and we display the error message to the console, should we ignore:
* just that command (and continue trying to execute subsequent commands)
* the erroneous command and all subsequent commands
* All commands, and do nothing at all?

Reply:
Should this default to the `"automatic"` split behavior?

Reply:
Yes it should default to auto. Auto hadn't even been proposed when i first wrote this, but that's a _very_ sensible default